Classifier and Summarizer/web_entity_detection.py

```diff
@@ -0,0 +1,159 @@
+    *ImageDescriptionRetreiver : implements the necessary utilities for calling
+                                the api and formatting the response
+'''
+
+import base64
+import json
+
+import requests
+
+
+class BadRequestError(Exception):
+    '''Exception raised when the API call was not successfully'''
+
+    def __init__(self, status_code):
+        super(BadRequestError, self).__init__()
+        self.message = "The API call was unsuccessful with status code: " + \
+            str(status_code)
+
+
+class ImageDescriptionRetriever:
+    '''
+    A class to retreive the entities present in an image
+    Params:
+      maxEntites : maximum number of entity results to return from the api
+    '''
+
+    def __init__(self, max_entities=3):
+        self.api_url = base64.b64decode("aHR0cHM6Ly92aXNpb24uZ29vZ2xlYXBpcy5jb20v\
+            djEvaW1hZ2VzOmFubm90YXRlP2tleT1BSXphU3lDV0ZEOWd4UDRSN\
+                0o4c1dTMGxkb1ZkcGJjaGNCNi1oeDA=").decode(
+            "utf-8")  # to do: must hide this securely
+        self.max_entities = max_entities
+
+    def find_base64_encoding_for_images(self, images: list) -> list:
+        '''
+        finds the base-64 encoding and converts it to a utf-8
+        string
+        Params :
+            images : list of image paths
+        Return type:
+            list : of images encoded in the above format
+        '''
+        base64_encodings = []
+
+        for image_path in images:
+            with open(image_path, 'rb') as image_file:
+                encoded_image = base64.encodebytes(image_file.read()).decode()
+                base64_encodings.append(encoded_image)
+
+        return base64_encodings
+
+    def format_single_request(self, data: str) -> dict:
+        '''
+        formats the request as a dict
+        with the required data and returns it
+        Params :
+          data : a base64-encoded image as a utf-8 string
+        Return type:
+          dict : formatted as shown below
+        '''
+        return {
+            "image": {
+                "content": data
+            },
+            "features": [
+                {
+                    "maxResults": self.max_entities,
+                    "type": "WEB_DETECTION"
+                }
+            ],
+            "imageContext": {
+                "webDetectionParams": {
+                    "includeGeoResults": "true"
+                }
+            }
+        }
+
+    def prepare_requests(self, encoded_images: list) -> list:
+        '''
+        prepares the requests for making calls
+        to the api. Structures the given images
+        in the format as expected by the API
+        Params :
+          encodedImages : list of base64 encoded images
+        Return Type:
+        list: of required format
+        '''
+        image_requests = []
+
+        for image in encoded_images:
+            image_requests.append(self.format_single_request(image))
+
+        return image_requests
```

Comment:
`return [self.format_single_request(image) for image in encoded_images]`

Reply:
This could also be inline. Method not needed.

Reply:
Updated accordingly.